### PR TITLE
[Implement] Buffer Read/Write 32 Bit Int/Float

### PR DESCRIPTION
**Pull request by RedDwarfian**

### CODE REVIEW

File: assembly/buffer/index.ts [hunk 1]
```diff
@@ -92,4 +92,70 @@ export class Buffer extends Uint8Array {
...
+
+  writeUInt32BE(value: u32, offset: i32 = 0): i32 {
+    if(i32(offset < 0) | i32(<u32>offset + 4 > this.dataLength)) throw new RangeError(E_INDEXOUTOFRANGE);
+    store<u32>(this.dataStart + offset, bswap<u32>(value));
+    return offset + 4;
+  }
+
+  readFloatLE(offset: i32 = 0): f32 {
+    if(i32(offset < 0) | i32(<u32>offset + 4 > this.dataLength)) throw new RangeError(E_INDEXOUTOFRANGE);
+    return reinterpret<f32>(load<i32>(this.dataStart + <usize>offset));
```
Comment (dcodeIO) [inline assembly/buffer/index.ts]: Can just `load<f32>` here

File: assembly/buffer/index.ts [hunk 2]
```diff
@@ -92,4 +92,70 @@ export class Buffer extends Uint8Array {
...
+  }
+
+  readFloatBE(offset: i32 = 0): f32 {
+    if(i32(offset < 0) | i32(<u32>offset + 4 > this.dataLength)) throw new RangeError(E_INDEXOUTOFRANGE);
+    return reinterpret<f32>(bswap<i32>(load<i32>(this.dataStart + <usize>offset)));
+  }
+
+  writeFloatLE(value: f32, offset: i32 = 0): i32 {
+    if(i32(offset < 0) | i32(<u32>offset + 4 > this.dataLength)) throw new RangeError(E_INDEXOUTOFRANGE);
+    store<i32>(this.dataStart + offset, reinterpret<i32>(value));
```
Comment (dcodeIO) [inline assembly/buffer/index.ts]: Can just `store<f32>` here

**Review by jtenner — APPROVED** [review]
This pull request looks excellent now.

@dcodeio?

**Comment by MaxGraey** [comment]
It seems this PR miss `readDoubleLE/readDoubleBE` and `writeDoubleLE/writeDoubleBE`?

**Comment by jtenner** [comment]
He told me he wanted to cover the 32 bit operations in this pull request. He already started work on those functions elsewhere.

**Comment by MaxGraey** [comment]
Got it

**Review by jtenner — CHANGES_REQUESTED** [review]
One last nitpick.

```ts
if (i32(offset < 0)... 
```

Can we put a space between the if and each parenthesis in the `assembly/buffer/index.ts` file?

Once this is done, I'll pull the request.

**Comment by MaxGraey** [comment]
LGTM

**Pull request merged**